### PR TITLE
Adding `TextFieldBase` component

### LINKED ISSUE #15099: [Ext Nav] Create component: TextFieldBase
### Description
Create a reusable UI component for `TextFieldBase`

### References
[Figma Link](https://www.figma.com/file/HKpPKij9V3TpsyMV1TpV7C/DS-Components?node-id=7267%3A28971)
[UI component guidelines](https://www.notion.so/UI-Components-5ccd2bf83eb8441892a0b72c0d8929e1)
[Styling guidelines](https://www.notion.so/Extension-Frontend-Engineering-Guide-b82ddb3e14004b5db3799a9b446294a9#49aa2ef7b7aa40178ff918043df71c34)
Testing guidelines: TBC

### Files needed
- `index.js`
- `index.scss`
- `text-field-base.js`
- `text-field.stories.js`
- `text-field.test.js`
- `README.mdx`

### Technical details
TBC

### Acceptance criteria
- [ ] Uses color, typography, shadows design tokens 
- [ ] Uses semantic html
- [ ] PropTypes have descriptions
- [ ] Has storybook story with controls
- [ ] Has documentation in MDX
- [ ] Has unit tests and 90% coverage
- [ ] Works in Chrome and Firefox
- [ ] Performance tested: no unnecessary re-renders or other performance concerns[

### PR DESCRIPTION
**Pull request by georgewrmarshall**
## Explanation

Adding `TextFieldBase` component as a base for all text type components. Borrowed some architectural decisions [from material ui](https://github.com/mui/material-ui/blob/78fc73ccb93628f0a7a762f1023de896c1be36f4/packages/mui-material/src/TextField/TextField.js)

### Architecture
The `TextFieldBase` component is made up of 2 components the wrapping `<Box />` and `<Text />` components. The Box contains all of the visual css and allows for right and left accessories to be placed outside of the html input element. The html input element is the root element of the Text component. In a previous PR I forwarded refs to both Box and Text to allow for focusing during the onClick on the wrapping Box component. This allows for the input to look focused while maintaining the left and right accessories to be focusable using the keyboard and maintaining accessibility

Screen Shot 2022-10-04 at 1 52 54 PM

https://user-images.githubusercontent.com/8112138/193925236-eeb91023-51ab-437a-bf84-4e7d01c75825.mov

## More Information
Fixes #15099 
[Figma link](https://www.figma.com/file/HKpPKij9V3TpsyMV1TpV7C/DS-Components?node-id=1244%3A11663)

## Screenshots/Screencaps

https://user-images.githubusercontent.com/8112138/193337278-96da5319-37dc-4f92-8402-97eb79b1bbe3.mov

## Manual Testing Steps

- Go to the latest build of storybook on this PR

## Pre-Merge Checklist

- [x] PR template is filled out
- [x] **IF** this PR fixes a bug, a test that _would have_ caught the bug has been added
- [x] PR is linked to the appropriate GitHub issue
- [x] PR has been added to the appropriate release Milestone

### + If there are functional changes:

- [x] Manual testing complete & passed
- [x] "Extension QA Board" label has been applied

### CODE REVIEW

File: ui/components/component-library/base-avatar/base-avatar.scss [hunk 1]
```diff
@@ -23,6 +23,8 @@
 
   height: var(--avatar-size);
   width: var(--avatar-size);
+  max-width: var(--avatar-size);
+  flex: 0 0 var(--avatar-size);
   border-radius: 100%;
```
Comment (georgewrmarshall) [inline ui/components/component-library/base-avatar/base-avatar.scss:28]: Will move this to a separate PR it fixes the display of the Avatar in a flex-box context with elements of different heights.

### Before

Screen Shot 2022-09-30 at 11 54 12 AM


### After

Screen Shot 2022-09-30 at 11 54 39 AM

File: ui/components/component-library/text-field-base/README.mdx [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+  <Story id="ui-components-component-library-text-field-base-text-field-base-stories-js--default-story" />
+</Canvas>
+
+## Props
+
+The `TextFieldBase` accepts all props below as well as all [Box](/docs/ui-components-ui-box-box-stories-js--default-story#props) component props
+
+<ArgsTable of={TextFieldBase} />
+
+<!-- TODO: Add all docs -->
```
Comment (georgewrmarshall) [inline ui/components/component-library/text-field-base/README.mdx:21]: Will add more docs just wanted to get some 👀 on this PR while I'm creating them

File: ui/components/component-library/text-field-base/text-field-base.js [hunk 3]
```diff
@@ -0,0 +1,285 @@
...
+    };
+
+    const handleInputRef = (ref) => {
+      internalInputRef.current = ref;
+      if (inputRef && inputRef.current !== undefined) {
+        inputRef.current = ref;
+      } else if (typeof inputRef === 'function') {
+        inputRef(ref);
+      }
+    };
```
Comment (georgewrmarshall) [inline ui/components/component-library/text-field-base/text-field-base.js:113]: Need to figure out my uncontrolled > controlled issue

Screen Shot 2022-09-30 at 11 57 58 AM

File: ui/components/component-library/base-avatar/base-avatar.scss (hunk 1 above)
Comment (georgewrmarshall) [reply ui/components/component-library/base-avatar/base-avatar.scss:28]: Fixed in https://github.com/MetaMask/metamask-extension/pull/16047

**Review by brad-decker — APPROVED** [review]
Awesome

**Review by garrettbear — APPROVED** [review]
